frb_example/pure_dart/rust/src/bridge_generated.io.rs

```diff
@@ -805,640 +904,933 @@ pub extern "C" fn new_uint_8_list_0(len: i32) -> *mut wire_uint_8_list {
     }
 }
 impl Wire2Api<String> for *mut wire_uint_8_list {
-    fn wire2api(self) -> String {
-        let vec: Vec<u8> = self.wire2api();
-        String::from_utf8_lossy(&vec).into_owned()
+    fn wire2api(self) -> Result<String, &'static str> {
+        let vec: Vec<u8> = self.wire2api()?;
+        Ok(String::from_utf8_lossy(&vec).into_owned())
     }
 }
 impl Wire2Api<Vec<String>> for *mut wire_StringList {
-    fn wire2api(self) -> Vec<String> {
+    fn wire2api(self) -> Result<Vec<String>, &'static str> {
         let vec = unsafe {
             let wrap = support::box_from_leak_ptr(self);
             support::vec_from_leak_ptr(wrap.ptr, wrap.len)
         };
-        vec.into_iter().map(Wire2Api::wire2api).collect()
+        vec.into_iter()
+            .map(Wire2Api::wire2api)
+            .collect::<Vec<Result<String, &str>>>()
+            .into_iter()
+            .collect()
     }
 }
 impl Wire2Api<[TestId; 4]> for *mut wire_list_test_id {
-    fn wire2api(self) -> [TestId; 4] {
-        let vec: Vec<TestId> = self.wire2api();
-        support::from_vec_to_array(vec)
+    fn wire2api(self) -> Result<[TestId; 4], &'static str> {
+        let vec: Vec<TestId> = self.wire2api()?;
+        Ok(support::from_vec_to_array(vec))
     }
 }
 impl Wire2Api<uuid::Uuid> for *mut wire_uint_8_list {
-    fn wire2api(self) -> uuid::Uuid {
-        let single: Vec<u8> = self.wire2api();
-        wire2api_uuid_ref(single.as_slice())
+    fn wire2api(self) -> Result<uuid::Uuid, &'static str> {
+        let single: Vec<u8> = self.wire2api()?;
+        Ok(wire2api_uuid_ref(single.as_slice()))
     }
 }
 impl Wire2Api<Vec<uuid::Uuid>> for *mut wire_uint_8_list {
-    fn wire2api(self) -> Vec<uuid::Uuid> {
-        let multiple: Vec<u8> = self.wire2api();
-        wire2api_uuids(multiple)
+    fn wire2api(self) -> Result<Vec<uuid::Uuid>, &'static str> {
+        let multiple: Vec<u8> = self.wire2api()?;
+        Ok(wire2api_uuids(multiple))
     }
 }
 impl Wire2Api<ZeroCopyBuffer<Vec<u8>>> for *mut wire_uint_8_list {
-    fn wire2api(self) -> ZeroCopyBuffer<Vec<u8>> {
-        ZeroCopyBuffer(self.wire2api())
+    fn wire2api(self) -> Result<ZeroCopyBuffer<Vec<u8>>, &'static str> {
+        Ok(ZeroCopyBuffer(self.wire2api()?))
     }
 }
 impl Wire2Api<ApplicationEnv> for wire_ApplicationEnv {
-    fn wire2api(self) -> ApplicationEnv {
-        ApplicationEnv {
-            vars: self.vars.wire2api(),
-        }
+    fn wire2api(self) -> Result<ApplicationEnv, &'static str> {
+        let vars = self.vars.wire2api();
+        Ok(ApplicationEnv { vars: vars? })
     }
 }
 impl Wire2Api<ApplicationEnvVar> for wire_ApplicationEnvVar {
-    fn wire2api(self) -> ApplicationEnvVar {
-        ApplicationEnvVar(self.field0.wire2api(), self.field1.wire2api())
+    fn wire2api(self) -> Result<ApplicationEnvVar, &'static str> {
+        let field0 = self.field0.wire2api();
+        let field1 = self.field1.wire2api();
+        Ok(ApplicationEnvVar(field0?, field1?))
```

Comment:
One problem if we make `wire2api` fall-able: What if multiple calls fail? For example, here, suppose field0 and field1 both fails. Then indeed we only throw (propagate) field0's error while secretly ignore field1's error.

If wire2api is fallible, maybe this is nontrivial to solve. For example, we have to do nontrivial changes to all generators such as enum, struct, ...

Reply:
I don't think this is a problem. This behavior is essentially an early return but with normal memory clearing

for example we would also ignore field1 error
```
Ok(ApplicationEnvVar(self.field0.wire2api()?, self.field1.wire2api()?))
```

Reply:
Hmm seems a bit different

```rs
use anyhow::*;

fn f() -> Result<i32> {
    println!("f called");
    Err(anyhow!("oops"))
}

fn g(a: i32, b: i32) -> i32 {a+b}

fn h() -> Result<i32> {
    let a = f();
    let b = f();
    Ok(g(a?, b?))
    
    //Ok(g(f()?, f()?))
}

fn main() {
    h();
}
```

* In this case, f is called twice.
* If use the commented out code, `Ok(g(f()?, f()?))`, f is called once.

In other words, when the first one throws error, the second one is never called at all

Reply:
That said, if this is a common practice (though have the drawback of sliently ignoring one error), such as there is some articles or docs, then we can use it

Reply:
(also see alternative solution mentioned below)

Reply:
This is exactly why we require wire2api to be called on each field. You think what we have indicated in the comments is not enough?

Reply:
I am just replying to "...for example we would also ignore field1 error...".

What do you think about the alternative solution in the comment below? That will eliminate the whole problem.

Reply:
This looks like the last blocking issue before merging, and I cannot wait to merge it!

(On the contrary, many of my PRs to Flutter gets rejected or overhaul: https://cjycode.com/flutter_smooth/insight/status)